cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -552,7 +552,17 @@ find_package(Python3 3.8 COMPONENTS ${python_components})
 #---Check for OpenGL installation-------------------------------------------------------
 if(opengl)
   message(STATUS "Looking for OpenGL")
-  find_package(OpenGL)
+  if(APPLE)
+    if(cocoa)
+      set(CMAKE_FIND_FRAMEWORK FIRST)
+    endif()
+    find_package(OpenGL)
+    if(cocoa)
+      set(CMAKE_FIND_FRAMEWORK LAST)
```

Comment:
Would it not be safer to reset it to the previous value as suggested in this PR?
https://github.com/root-project/root/pull/15435/files

Great that you figured out that `FIRST` is the right setting to find `OpenGL`! I tried `ONLY` back then, but the CI was not green for whatever reason.

Reply:
`ONLY` should work without problems. I tested on my two system (arm and x86) and there it worked. 
I would alos propose to save the previous setting and set it back after finding OpenGL.

Reply:
The previous setting is `LAST`  set at line 23:
https://github.com/root-project/root/blob/master/cmake/modules/SearchInstalledSoftware.cmake

Reply:
I think we can do:
```
 if(APPLE)
      set(CMAKE_FIND_FRAMEWORK FIRST)
      find_package(OpenGL)
      set(CMAKE_FIND_FRAMEWORK LAST)
  else()
    find_package(OpenGL)
  endif()
```
that was my first idea.